web-marketplace/src/apollo.tsx

```diff
@@ -54,6 +67,42 @@ export const AuthApolloProvider = ({
     },
   };
 
+  const errorLink = onError(({ operation, networkError, forward }) => {
+    if (
+      networkError &&
+      networkError.message.includes('status code 403') &&
+      operation.operationName !== 'GetCurrentAccount'
+    ) {
+      const observable = new Observable<FetchResult<Record<string, any>>>(
+        observer => {
+          (async () => {
+            try {
+              await reactQueryClient.invalidateQueries({
+                queryKey: [GET_CSRF_QUERY_KEY],
+              });
+              await getFromCacheOrFetch({
```

Comment:
why do we need this `getFromCacheOrFetch` too?

Reply:
to trigger the call and refill the cache earlier.

Reply:
It would work without it, it's just a network optimization.

Reply:
ok could you add a comment about that? thanks